drivers/clock_control/Kconfig.mchp

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2025 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Symbol for enabling clock driver for a family
+config CLOCK_CONTROL_MCHP_SAM_D5X_E5X
+	bool
+	default y
+	depends on DT_HAS_MICROCHIP_SAM_D5X_E5X_CLOCK_ENABLED
+	help
+	  This symbol define clock driver for Microchip SAM_D5X_E5X family.
+
+# Define menu symbol for configuring clock features
+menuconfig CLOCK_CONTROL_MCHP
+	bool "Microchip clock controller driver"
+	default y
+	depends on CLOCK_CONTROL_MCHP_SAM_D5X_E5X
+	help
+	  Enable support for Microchip clock controller driver.
```

Comment:
this is superfluous
```suggestion
menuconfig CLOCK_CONTROL_MCHP
	bool "Microchip clock controller driver"
	default y
	depends on DT_HAS_MICROCHIP_SAM_D5X_E5X_CLOCK_ENABLED
	help
	  Enable support for Microchip clock controller driver.
```

Reply:
can have symbols like `CLOCK_CONTROL_MCHP_SAM_D5X_E5X` (which has the depends on DT_...) select a Kconfig with no prompt e.g. `CLOCK_CONTROL_MCHP_COMMON` then have options displayed there

Reply:
```suggestion
config CLOCK_CONTROL_MCHP_COMMON
        bool
        help
          Common option for microchip clock control drivers

config CLOCK_CONTROL_MCHP_SAM_D5X_E5X
        bool "Text here"
        depends on DT_...
        select CLOCK_CONTROL_MCHP_COMMON
```